### PR TITLE
Add some simple tests for nvidia GPU(s)

### PR DESCRIPTION
**Pull request by dims**
xref: https://github.com/kubernetes/kubernetes/issues/124950

Add some really basic sanity tests for nvidia, starting with AWS as provider here (so it will show up in https://testgrid.k8s.io/amazon-ec2#ci-kubernetes-e2e-ec2-device-plugin-gpu&width=20) without breaking other CI jobs.

- Run nvidia-smi CLI and check output
- Run a tensorflow image and do a matrix multiplication 

/kind regression

(Marking as regression because we lost GPU test coverage with the cloud provider migration)

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by dims** [comment]
cc @BenTheElder @SergeyKanzhelev

/assign @hakuna-matatah

File: test/e2e/node/gpu.go [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+	f.NamespacePodSecurityLevel = admissionapi.LevelPrivileged
+	var podClient *e2epod.PodClient
+
+	ginkgo.BeforeEach(func() {
+		e2eskipper.SkipUnlessProviderIs("aws")
+	})
+
+	ginkgo.BeforeEach(func() {
+		podClient = e2epod.NewPodClient(f)
+	})
```
Comment (hakuna-matatah) [inline test/e2e/node/gpu.go:50]: We can club this into one ?

File: test/e2e/node/gpu.go [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+					},
+				},
+			},
+			RestartPolicy: v1.RestartPolicyNever,
+		},
+	}
+	return &pod
+}
+
+func testMatrixMultiplicationPod() *v1.Pod {
```
Comment (hakuna-matatah) [inline test/e2e/node/gpu.go:141]: May be in future we can combine [this](https://github.com/kubernetes/kubernetes/pull/127423/files#diff-473979ff046f836345a65ff99dc9fbb3a0e6141232f2413eb296eb7f8c920b2eR116) func and this function into more of a generic function something like :

`func createPod(namePrefix, image string, command []string) *v1.Pod {
...
}
`
 ?

File: test/e2e/node/gpu.go [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+
+		ginkgo.By("Waiting for pod completion")
+		err = e2epod.WaitForPodNoLongerRunningInNamespace(ctx, f.ClientSet, pod.Name, f.Namespace.Name)
+		framework.ExpectNoError(err)
+		pod, err = podClient.Get(ctx, pod.Name, metav1.GetOptions{})
+		framework.ExpectNoError(err)
+
+		ginkgo.By("Checking that the pod succeeded")
+		gomega.Expect(pod.Status.Phase).To(gomega.Equal(v1.PodSucceeded))
+
```
Comment (hakuna-matatah) [inline test/e2e/node/gpu.go:73]: May be in next iteration we can move this to its own function ? As we are repeating these steps for other NVDIA pod [here](https://github.com/kubernetes/kubernetes/pull/127423/files#diff-473979ff046f836345a65ff99dc9fbb3a0e6141232f2413eb296eb7f8c920b2eR88-R104)

`func createAndValidatePod(ctx context.Context, f *framework.Framework, pod *v1.Pod) *v1.Pod {
... }
`

File: test/e2e/node/gpu.go (hunk 1 above)
Comment (dims) [reply test/e2e/node/gpu.go:50]: Done!

File: test/e2e/node/gpu.go (hunk 2 above)
Comment (dims) [reply test/e2e/node/gpu.go:141]: yep, we can do that in a follow up

File: test/e2e/node/gpu.go (hunk 3 above)
Comment (dims) [reply test/e2e/node/gpu.go]: Done!

**Comment by hakuna-matatah** [comment]
/lgtm

Thank you for doing this.

File: test/e2e/node/gpu.go [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+					},
+				},
+			},
+			RestartPolicy: v1.RestartPolicyNever,
+		},
+	}
+	return &pod
+}
+
+func checkEnvironmentAndSkipIfNeeded(ctx context.Context, clientSet clientset.Interface) {
```
Comment (SergeyKanzhelev) [inline test/e2e/node/gpu.go:183]: Sorry, I cannot find where is the driver being installed. Or we assume the environment has a driver installed?

Comment (SergeyKanzhelev) [reply test/e2e/node/gpu.go:183]: /hold

to understand what we test

Comment (dims) [reply test/e2e/node/gpu.go:183]: @SergeyKanzhelev `kubetest2 ec2` has an option
https://cs.k8s.io/?q=DevicePluginNvidia%7CENABLE_NVIDIA_DEVICE_PLUGIN&i=nope&files=&excludeFiles=&repos=

Comment (dims) [reply test/e2e/node/gpu.go:183]: xref: https://github.com/kubernetes/test-infra/pull/33492

Comment (SergeyKanzhelev) [reply test/e2e/node/gpu.go:183]: I wonder if we need to rely on device driver that comes with the platform or should have it installed by test. If installed by test - we will have more explicit statement on what nvidia drivers k8s is compatible with.

Comment (dims) [reply test/e2e/node/gpu.go:183]: @SergeyKanzhelev i am happy whichever way you want to go with. The main focus of this PR to get something up which we can then tweak as we are blind right now. Once i get this working on AWS, we can figure out whether we want to do the GCP one via a change in kubeup or e2e-test code and then i can fix up aws side to match.

Comment (SergeyKanzhelev) [reply test/e2e/node/gpu.go:183]: ok, makes sense

/unhold

Comment (dims) [reply test/e2e/node/gpu.go:183]: thanks @SergeyKanzhelev

**Pull request merged**